src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,37 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(target) {
+    if (target instanceof Herbivore && !target.hidden) {
+      target.health -= 50;
+    }
+
+    if (target.health === 0) {
```

Comment:
What if there will be less than 0?

Reply:
We are deleting animal with health > 0, so we never get into this case. or not?

Reply:
if we have some variables for (-50) - ok, I agree with you. But here we have to use directly -50. So it's impossible to go to the case health < 0.

Reply:
Good point, you are correct. But for the future, it is safer to use `<= 0`